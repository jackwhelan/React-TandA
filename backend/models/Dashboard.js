const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const DashboardSchema = new Schema({
    clearance: {
        type: String
    },
    dashboardItems: {
        type: [Object]
    }
});

const Dashboard = new mongoose.model('Dashboard', DashboardSchema);


module.exports = Dashboard;