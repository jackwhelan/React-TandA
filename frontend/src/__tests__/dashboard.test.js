import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Dashboard from '../components/dashboard/Dashboard.component';
import DashboardItem from '../components/dashboard/DashboardItem.component';

describe('Dashboard Components', () => {
    test('Dashboard', () => {
        const component = renderer.create(<Router><Dashboard /></Router>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Dashboard Item', () => {
        const component = renderer.create(<DashboardItem />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})