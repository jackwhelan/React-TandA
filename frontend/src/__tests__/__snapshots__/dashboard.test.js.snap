// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Components Dashboard 1`] = `null`;

exports[`Dashboard Components Dashboard Item 1`] = `
<a
  className="link-unstyled"
>
  <div
    className="row"
  >
    <div
      className="border border-dark col-sm-4 p-3 margin-dash"
    >
      <h5
        className="text-center scaleIcon"
      >
        <i />
      </h5>
    </div>
    <div
      className="border border-dark col-sm-7 p-3 margin-dash"
    >
      <h4 />
      <p />
    </div>
  </div>
</a>
`;
